repo/blob/azure/azure_storage.go

```diff
@@ -278,20 +280,26 @@ func (az *azStorage) retryDeleteBlob(ctx context.Context, b blob.ID) error {
 		return errors.Wrap(err, "failed to soft delete blob")
 	}
 
+	log := logging.Module("azure-immutability")
+
 	if resp.VersionID == nil || *resp.VersionID == "" {
 		// shouldn't happen
+		log(ctx).Info("VersionID not returned, exiting without deleting the delete marker version")
 		return nil
 	}
+
 	bc, err := az.service.ServiceClient().
 		NewContainerClient(az.container).
 		NewBlobClient(blobName).
 		WithVersionID(*resp.VersionID)
 	if err != nil {
+		log(ctx).Infof("Issue preparing versioned blob client: %v", err)
```

Comment:
isn't this going to require a blank line before the `return` statement?
Same below.

Reply:
Apparently not.